js-miniapp-sdk/src/miniapp-error-type.ts

```diff
@@ -0,0 +1,99 @@
+interface MiniAppJson {
+  message: string;
+  type: string;
+}
+
+export function parseMiniAppError(jsonString: string): MiniAppJson {
+  return JSON.parse(jsonString);
+}
+
+/**
+ * This class is a representation of an error sent from MiniApp mobile SDK
+ */
+export class MiniAppError extends Error {
+  /**
+   * the raw input received from the SDK used to construct the MiniAppError
+   */
+  raw: string;
+  customMessage: string;
+
+  constructor(public errorInput: MiniAppJson) {
```

Comment:
See here in the firebase repo: 
https://github.com/firebase/firebase-js-sdk/blob/5ad7ff2ae955c297556223e6cb3ad9d4b897f664/packages/util/src/errors.ts#L86

You have to set `Object.setPrototypeOf(this, MiniAppError.prototype)` or else I don't think you can use `instanceof` to check the type. I think it's some weird quirk of TypeScript.

Also, you need to set it in each sub-type: `Object.setPrototypeOf(this, AudienceNotSupportedError.prototype)`, etc.